runtime/lib/ttnn/include/tt/runtime/ttnn/types.h

```diff
@@ -161,6 +167,14 @@ class ProgramContext {
   // Tensor Pool Operations
   //
   ProgramTensorPool &getTensorPool() { return tensorPool; }
+  const ProgramTensorPool &getTensorPool() const { return tensorPool; }
+
+  //
+  // Golden Tensor Operations
+  //
+  const std::unordered_map<std::string, Tensor> &getGoldenMap() {
```

Comment:
If this is a debug feature I don't think we would want to store this in the program context. Is there a way to separate this from the main runtime flow into its own infra?